python3Packages.oca-port: init at 0.13

yajo opened the pull request:
###### Description of changes
Useful CLI tool for [OCA](https://odoo-community.org/) modules maintenance.

@moduon MT-2586

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

natsukium commented on pkgs/development/python-modules/oca-port/default.nix:
```diff
@@ -0,0 +1,45 @@
+    src = fetchPypi {
+      inherit version;
+      pname = moduleName;
+      sha256 = "sha256-adwqnhe5BLo5tsjaLb0XHlTFxSKNutsb6hlTXjA4Wio=";
```
```suggestion
      hash = "sha256-adwqnhe5BLo5tsjaLb0XHlTFxSKNutsb6hlTXjA4Wio=";
```
The SRI hash is preferred.

natsukium commented on pkgs/development/python-modules/oca-port/default.nix:
```diff
@@ -0,0 +1,45 @@
+
+    passthru.updateScript = nix-update-script { };
+
+    pythonImportsCheck = [moduleName];
```
```suggestion
    pythonImportsCheck = [ moduleName ];
```

natsukium commented on pkgs/development/python-modules/oca-port/default.nix:
```diff
@@ -0,0 +1,45 @@
+    pythonImportsCheck = [moduleName];
+
+    meta = with lib; {
+      description = "Tool helping to port an addon or missing commits of an addon from one branch to another.";
```
```suggestion
      description = "Tool helping to port an addon or missing commits of an addon from one branch to another";
```

natsukium commented on pkgs/development/python-modules/oca-port/default.nix:
```diff
@@ -0,0 +1,45 @@
+    meta = with lib; {
+      description = "Tool helping to port an addon or missing commits of an addon from one branch to another.";
+      homepage = "https://github.com/OCA/oca-port";
+      license = licenses.lgpl3;
```
```suggestion
      license = licenses.lgpl3Plus;
```

natsukium commented on pkgs/development/python-modules/oca-port/default.nix:
```diff
@@ -0,0 +1,45 @@
+      description = "Tool helping to port an addon or missing commits of an addon from one branch to another.";
+      homepage = "https://github.com/OCA/oca-port";
+      license = licenses.lgpl3;
+      maintainers = [maintainers.yajo];
```
```suggestion
      maintainers = [ maintainers.yajo ];
```

natsukium commented on pkgs/development/python-modules/oca-port/default.nix:
```diff
@@ -0,0 +1,45 @@
+  buildPythonPackage,
+  click,
+  fetchPypi,
+  GitPython,
```
gitpython?

yajo replied:
It's a dependency: https://github.com/OCA/oca-port/blob/cebdc50f06268500783ec5667d6780e6a549d3a9/pyproject.toml#L18

natsukium replied:
Yes, GitPython is `gitpython` in nixpkgs.
This causes the build to fail.

yajo replied:
Ah sorry, it was building locally and that's why it was confusing to me. Changed now.

yajo commented:
All attended, thanks! I forgot to run nixpkgs-fmt over the file. It should be fine now.

yajo committed:
python3Packages.oca-port: init at 0.12

Useful CLI tool for [OCA](https://odoo-community.org/) modules maintenance.

@moduon MT-2586

Co-authored-by: 25083790+natsukium@users.noreply.github.com
Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

yajo commented:
AFAICS it all should be fixed now.

natsukium commented:
Result of `nixpkgs-review pr 224626` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.oca-port</li>
    <li>python310Packages.oca-port.dist</li>
    <li>python311Packages.oca-port</li>
    <li>python311Packages.oca-port.dist</li>
  </ul>
</details>

natsukium reviewed (approved):
LGTM, thanks!

yajo commented:
So, could we merge?

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2275

SuperSandro2000 commented on pkgs/development/python-modules/oca-port/default.nix:
```diff
@@ -0,0 +1,44 @@
+  meta = with lib; {
+    description = "Tool helping to port an addon or missing commits of an addon from one branch to another";
+    homepage = "https://github.com/OCA/oca-port";
+    license = licenses.lgpl3Plus;
```
meta.maintainers must be set

SuperSandro2000 commented on pkgs/development/python-modules/oca-port/default.nix:
```diff
@@ -0,0 +1,44 @@
+, requests
+, setuptools-scm
+, setuptools
+}:
+let
+  moduleName = "oca_port";
+in
+buildPythonPackage rec {
```
```suggestion
}:

buildPythonPackage rec {
```

SuperSandro2000 commented on pkgs/development/python-modules/oca-port/default.nix:
```diff
@@ -0,0 +1,44 @@
+
+  passthru.updateScript = nix-update-script { };
+
+  pythonImportsCheck = [ moduleName ];
```
```suggestion
  pythonImportsCheck = [ "oca_port" ];
```

SuperSandro2000 commented on pkgs/development/python-modules/oca-port/default.nix:
```diff
@@ -0,0 +1,44 @@
+  version = "0.12";
+
+  src = fetchPypi {
+    inherit version;
+    pname = moduleName;
```
```suggestion
    pname = "oca_port",
    inherit version;
```

SuperSandro2000 commented on pkgs/development/python-modules/oca-port/default.nix:
```diff
@@ -0,0 +1,44 @@
+in
+buildPythonPackage rec {
+  pname = "oca-port";
+  version = "0.12";
```
```suggestion
  version = "0.12";
  format = "pyproject";
```

SuperSandro2000 commented on pkgs/development/python-modules/oca-port/default.nix:
```diff
@@ -0,0 +1,44 @@
+  src = fetchPypi {
+    inherit version;
+    pname = moduleName;
+    hash = "sha256-adwqnhe5BLo5tsjaLb0XHlTFxSKNutsb6hlTXjA4Wio=";
+  };
+  buildInputs = [
+    setuptools
+    setuptools-scm
+  ];
+  propagatedBuildInputs = [
+    click
+    gitpython
+    requests
+  ];
+  format = "pyproject";
+  SETUPTOOLS_SCM_PRETEND_VERSION = version;
```
```suggestion
    hash = "sha256-adwqnhe5BLo5tsjaLb0XHlTFxSKNutsb6hlTXjA4Wio=";
  };

  nativeBuildInputs = [
    setuptools
    setuptools-scm
  ];

  propagatedBuildInputs = [
    click
    gitpython
    requests
  ];

  SETUPTOOLS_SCM_PRETEND_VERSION = version;
```

yajo commented:
All attended.

yajo committed:
python3Packages.oca-port: 0.12 -> 0.13

SuperSandro2000 commented:
@ofborg build python310Packages.oca-port
